src/mlpack/bindings/cli/add_to_cli11.hpp

```diff
@@ -0,0 +1,211 @@
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_BINDINGS_CLI_ADD_TO_PO_HPP
+#define MLPACK_BINDINGS_CLI_ADD_TO_PO_HPP
+
+#include <mlpack/core/util/param_data.hpp>
+#include <mlpack/core/util/is_std_vector.hpp>
+#include "map_parameter_name.hpp"
+
+#include "third_party/CLI/CLI11.hpp"
+
+namespace mlpack {
+namespace bindings {
+namespace cli {
+
+/**
+ * Add a tuple option to CLI11.
+ * 
+ * @param cliName The name of the option to add to CLI11.
+ * @param descr Description string for parameter.
+ * @param desc Options description to add parameter to.
+ */
+template<typename T>
+void AddToCLI11(const std::string& cliName,
+                util::ParamData& param,
+                CLI::App& app,
+                const typename boost::disable_if<std::is_same<T,
+                    bool>>::type* = 0,
+                const typename boost::disable_if<
+                    arma::is_arma_type<T>>::type* = 0,
+                const typename boost::disable_if<
+                    data::HasSerialize<T>>::type* = 0,
+                const typename boost::enable_if<std::is_same<T,
+                    std::tuple<mlpack::data::DatasetInfo,
+                    arma::mat>>>::type* = 0)
+{
+  app.add_option_function<std::string>(cliName.c_str(),
+      [&param](const std::string& value)
+      {
+        using TupleType = std::tuple<T, typename ParameterType<T>::type>;
+        TupleType& tuple = *boost::any_cast<TupleType>(&param.value);
+        std::get<1>(tuple) = boost::any_cast<std::string>(value);
+        param.wasPassed = true;
+      },
+      param.desc.c_str());
+}
+
+/**
+ * Add a serializable option to CLI11.
+ * 
+ * @param cliName The name of the option to add to CLI11.
+ * @param descr Description string for parameter.
+ * @param desc Options description to add parameter to.
+ */
+template<typename T>
+void AddToCLI11(const std::string& cliName,
+                util::ParamData& param,
+                CLI::App& app,
+                const typename boost::disable_if<std::is_same<T,
+                    bool>>::type* = 0,
+                const typename boost::disable_if<
+                    arma::is_arma_type<T>>::type* = 0,
+                const typename boost::enable_if<
+                    data::HasSerialize<T>>::type* = 0,
+                const typename boost::disable_if<std::is_same<T,
+                    std::tuple<mlpack::data::DatasetInfo,
+                    arma::mat>>>::type* = 0)
+{
+  app.add_option_function<std::string>(cliName.c_str(),
+      [&param](const std::string& value)
+      {
+        using TupleType = std::tuple<T*, typename ParameterType<T>::type>;
+        TupleType& tuple = *boost::any_cast<TupleType>(&param.value);
+        std::get<1>(tuple) = boost::any_cast<std::string>(value);
```

Comment:
I was looking at the Windows build failure.  I am not sure if this is the problem, but I did notice that in this function we forgot to set the parameter as `passed`.  If you add this line to the lambda I think it should help:

```
param.wasPassed = true;
```

Hopefully it will fix the Windows issue too.

Reply:
Exactly, only windows test are failing